lib/foreman_maintain/utils/backup.rb

```diff
@@ -241,6 +241,24 @@ def validate_hostname?
         end
       end
 
+      def validate_interfaces?
+        # I wanted to do `Socket.getifaddrs.map(&:name).uniq`,
+        # but this has to work with Ruby 2.0, and Socket.getifaddrs is 2.1+
+        system_interfaces = Dir.children('/sys/class/net')
+
+        proxy_config = metadata.fetch('proxy_config', {})
+
+        if proxy_config.fetch('dns', false) &&
+           !system_interfaces.include?(proxy_config.fetch('dns_interface', 'lo'))
+          return false
+        elsif proxy_config.fetch('dhcp', false) &&
+              !system_interfaces.include?(proxy_config.fetch('dhcp_interface', 'lo'))
+          return false
```

Comment:
DNS and DHCP could run on separate interfaces. I'd prefer to check both (so 2 if statements instead of elsif).